prifi-lib/client/client_test.go

```diff
@@ -701,19 +701,14 @@ func TestDisruptionClient(t *testing.T) {
 		dcNetDecoded[i] = pad1.Payload[i] ^ pad2.Payload[i] ^ clientPad.Payload[i]
 		i++
 	}
-	log.Error("dcNetDecoded", dcNetDecoded)
-
-	hmac := dcNetDecoded[0:32]
-	data := dcNetDecoded[32:]
-
-	success := relay.ValidateHmac256(data, hmac, clientID)
-	if !success {
-		t.Error("HMAC should be valid")
+	b_echo_last := dcNetDecoded[0]
+	if b_echo_last != 0 {
+		t.Error("The first b_echo_last flag should always be 0, now its:", b_echo_last)
 	}
 
 	// now in a normal round (possibly the client will send 0's but it's a different path)
-	//Receive some data down
-	dataDown := []byte{1, 2, 3}
+	// Receive some data down which is the CORRECT HASH of the previous message + [1, 2, 3]
+	dataDown := []byte{107, 219, 188, 254, 65, 196, 247, 119, 151, 176, 195, 248, 159, 252, 197, 57, 155, 53, 5, 34, 114, 223, 170, 75, 157, 208, 187, 59, 93, 214, 67, 191, 1, 2, 3}
```

Comment:
Let's avoid hardcoded magic values. This is the hash of what value ?
If it's the hash of the null slice, or a single 0 bit for instance, recompute it from from plaintext so it's clear what this value refers to